docs/startup.md

```diff
@@ -0,0 +1,30 @@
+# Metrictank startup
+
+The full startup procedure has many details, but here we cover the main steps if they affect:
+
+* performance/resource usage characteristics
+* cluster status
+* API availability
+* diagnostics
+
+
+| Phase                   | Description                                                                                        | effect on CPU / RAM                 |
+| ----------------------- | -------------------------------------------------------------------------------------------------- | ----------------------------------- |
+| load config             | load/validate config                                                                               | no                                  |
+| setup diagnostics       | set up logging, profiling, proftrigger                                                             | no                                  |
+| log startup             | logs "Metrictank starting" message                                                                 | no                                  |
+| start sending stats     | starts connecting and writing to graphite endpoint                                                 | no                                  |
+| create Store            | create keyspace, tables, write queues, etc                                                         | minor RAM increase ~ queue size     |
+| create Input(s)         | open connections (kafka) or listening sockets (carbon, prometheus)                                 | no                                  |
+| start cluster           | starts gossip, joins cluster                                                                       | no                                  |
+| create Index            | creates instance and starts write queues                                                           | minor RAM increase ~ queue size     |
+| start API server        | opens listening socket and starts handling requests                                                | no                                  |
+| init Index              | creates session, keyspace, tables, write queues, etc and loads in-memory index from persisted data | reasonable RAM and CPU increase                    |
+| create cluster notifier | optional: connects to Kafka, starts backfilling persistence message and waits until done or timeout| if backfilling: above-normal CPU, normal RAM usage |
+| start input plugin(s)   | starts backfill (kafka) or listening (carbon, prometheus) and maintain priority based on input lag | if backfilling: above-normal CPU and RAM usage     |
+| mark ready state        | immediately (primary) or after warmup period (secondary) (combined with priority for clustering)   | no                                                 |
+
+The CPU increase during the kafka backfilling is very significant. typically 10x compared to normal usage.
```

Comment:
I think `typically 10x` is a statement which includes many assumptions. In our case it might be `10x` but in other cases i think that number could be completely different. I'd rather articulate that in a way which does not make the reader assume that in their own environment it will probably also be `10x`. Something like
```
It can easily be 10x of the normal usage.
```

Reply:
our sizing recommendations (admittedly, not well documented) account for  being able to replay a ~7hr backlog in 30minutes, so 10~15x is expected for a properly set up ("typical") cluster. could be higher if the cluster is overprovisioned or lightly loaded, maybe i can clarify that.